### PR TITLE
Adding check for https protocol for repositories

### PR DESCRIPTION
**Pull request by palikar**
This should fix the problem outlined in #101

### CODE REVIEW

File: all_repos/push/github_pull_request.py [hunk 1]
```diff
@@ -12,13 +14,18 @@ class Settings(NamedTuple):
     username: str
     fork: bool = False
     base_url: str = 'https://api.github.com'
+    git_https_re: Pattern[str] = re.compile(r'http?s:\/\/.*?\/(.*)')
```
Comment (asottile) [inline all_repos/push/github_pull_request.py]: I don't think you need to escape forward slashes in regexes (only in javascript) -- also I think this is supposed to be `https?` ? (right now it matches `htts://`)

maybe also `[^/]+/(.*)$` would be slightly better for the matching portion too

**Comment by asottile** [comment]
actually, looking at this again I don't see how this fixes your problem (or how a user would configure the settings you've added).  `all-repos` also always generates repositories with ssh remotes and I don't think this changes that?

**Comment by palikar** [comment]
@asottile 
This maybe comes a little bit late but oh well. I fixed issues and tired adding a test for the new branch in the code. I am not sure if I've done it "properly" and I am pretty user it's kind of hacky.

**Comment by palikar** [comment]
> actually, looking at this again I don't see how this fixes your problem (or how a user would configure the settings you've added). `all-repos` also always generates repositories with ssh remotes and I don't think this changes that?

oh, ok, I  though that the remote URL can have two forms depending on how the repo was cloned. Does that mean that #101 is not an actual issue?

**Comment by asottile** [comment]
I'm not sure exactly -- it's certainly possible that someone writes an all-repos plugin which produces repositories with other protocols -- but none of the currently known plugins do so

**Pull request closed without merging**